arches/app/models/graph.py

```diff
@@ -485,6 +485,9 @@ def save(self, validate=True, nodeid=None):
                 self.update_es_node_mapping(node, datatype_factory, se)
                 self.create_node_alias(node)
                 try:
+                    if node.nodegroup:
+                        node.nodegroup.save()
```

Comment:
It seems like this call (and the same at ln 506) would result in multiple SQL statements to save the same nodegroup (ie it would save a nodegroup n times where n is the number of nodes in that group).  Can we adjust this logic so that it isn't adding this unnecessary SQL overhead?